protobuf/src/main/java/io/grpc/protobuf/StatusProto.java

```diff
@@ -158,7 +158,7 @@ private static Metadata toMetadata(com.google.rpc.Status statusProto, Metadata m
 
   @Nullable
   private static com.google.rpc.Status toStatusProto(Status status, Metadata trailers) {
-    if (trailers != null) {
+    if (trailers != null && trailers.containsKey(STATUS_DETAILS_KEY)) {
       com.google.rpc.Status statusProto = trailers.get(STATUS_DETAILS_KEY);
       checkArgument(
```

Comment:
The check for null should go here.  metadata uses a lot of linear time accesses, so doing it twice is reaching.